### PR TITLE
add php-ts-mode to auto-mode-alist when tree-sitter-php is present

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
to align with code from the core

### CODE REVIEW

File: php-ts-mode.el [hunk 1]
```diff
@@ -276,5 +276,9 @@ Return nil if there is no name or if NODE is not a defun node."
 
   (treesit-major-mode-setup))
 
+(when (treesit-ready-p 'php)
+  (add-to-list 'auto-mode-alist '("/\\.php_cs\\(?:\\.dist\\)?\\'" . php-ts-mode))
```
Comment (zonuexe) [inline php-ts-mode.el]: `.php_cs` is the configuration file of the old version of PHP-CS-Fixer, so I don't think it's necessary to support it newly, but what do you think?

Please see <https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/UPGRADE-v3.md>.

Comment (piotrkwiecinski) [reply php-ts-mode.el]: You're right. CS fixer 3 was released 2 years ago so I reckon it should be fine to remove it. I'll do it soon.

File: php-ts-mode.el [hunk 2]
```diff
@@ -276,5 +276,9 @@ Return nil if there is no name or if NODE is not a defun node."
...
+  (add-to-list 'auto-mode-alist '("\\.\\(?:php\\.inc\\|stub\\)\\'" . php-ts-mode)))
```
Comment (zonuexe) [inline php-ts-mode.el]: The [`.stub` extension seems to be used by languages other than PHP](https://github.com/search?q=path%3A*.stub&type=code), so we probably shouldn't add it by default.

```suggestion
  (add-to-list 'auto-mode-alist '("\\.\\(?:php[s345]?\\|phtml\\)\\'" . php-ts-mode) t))
```

I think it's fine to do this, but what do you think?

To prioritize `php-ts-mode` over `php-mode` from `php-mode-maybe` set as follows:

```el
(custom-set-variables
 '(php-default-major-mode 'php-ts-mode))
```

Comment (piotrkwiecinski) [reply php-ts-mode.el]: To be honest for start I think maybe we should just go with php[s345]? only as mixed type with html maybe tricker.

**Comment by zonuexe** [comment]
@piotrkwiecinski Thank you, I may replace it with append later after fixing `php-mode-maybe`.

**Pull request merged**